source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -804,26 +810,45 @@ def fill_journeys(self, request_type, api_request, instance):
         if not api_request['destination_mode']:
             api_request['destination_mode'] = ['walking']
 
+        # Return the possible couples combinations (origin_mode and destination_mode)
         krakens_call = get_kraken_calls(api_request)
 
+        # min_nb_journeys option
+        if api_request['min_nb_journeys']:
+            min_nb_journeys = api_request['min_nb_journeys']
+        else:
+            min_nb_journeys = api_request['min_nb_journeys'] = 1
+
+        # We need the original request (api_request) for filtering, but request
+        # is modified by create_next_kraken_request function.
         request = deepcopy(api_request)
-        min_asked_journeys = get_or_default(request, 'min_nb_journeys', 1)
+
         min_journeys_calls = get_or_default(request, '_min_journeys_calls', 1)
 
         responses = []
         nb_try = 0
-        nb_qualified_journeys = nb_journeys(responses)
+        nb_qualified_journeys = 0
 
         while request is not None and \
-                ((nb_qualified_journeys < min_asked_journeys and nb_try < min_asked_journeys)
-                 or nb_try < min_journeys_calls):
+                ((nb_qualified_journeys < min_nb_journeys and nb_try < min_nb_journeys) \
+                or nb_try < min_journeys_calls):
+
             nb_try = nb_try + 1
 
+            # we take into account the option only if we have one origin_mode and destination_mode couple.
+            if len(krakens_call) > 1:
+                request['min_nb_journeys'] = 0
+            else:
+                min_nb_journeys_left = min_nb_journeys - nb_qualified_journeys
+                request['min_nb_journeys'] = max(0, min_nb_journeys_left)
```

Comment:
The `max` is necessary. If kraken returns more itineraries than expected, `min_nb_journeys_left` will be negative because of the difference and we having a negative `min_nb_journey` doesn't make sense.

Reply:
:heavy_check_mark: yep anyway that's fine :)